src/Components/Subscriptions/Table/index.jsx

```diff
@@ -21,7 +23,12 @@ const Table = ({ data }) => {
       );
       setTableData(filteredData);
     }
-  }, [data, deletedSubscription]);
+    if (showDeleteModal) {
+      setTimeout(() => {
+        setShowDeleteModal(false);
+      }, 3000);
+    }
+  }, [data, deletedSubscription, showDeleteModal]);
 
   const handleEdit = (subscriptionId) => {
```

Comment:
we don't need to handle the edit anymore from here, if you agree, you can just take it out